contracts/basic-deal-client/DealClient.sol

```diff
@@ -63,64 +98,72 @@ contract DealClient {
+        ret.label = deal.label;
+        ret.start_epoch = deal.start_epoch;
+        ret.end_epoch = deal.end_epoch;
+        ret.storage_price_per_epoch = uintToBigInt(deal.storage_price_per_epoch);
+        ret.provider_collateral = uintToBigInt(deal.provider_collateral);
+        ret.client_collateral = uintToBigInt(deal.client_collateral);
+
+        return serializeDealProposal(ret);
     }
 
+    function getExtraParams(bytes32 proposalId) view public returns (bytes memory extra_params) {
+        // TODO make these array accesses safe.
+        DealRequest memory deal = deals[dealProposals[proposalId]];
+        return serializeExtraParamsV1(deal.extra_params);
+    }
 
     function authenticateMessage(bytes memory params) view internal {
-
         AccountTypes.AuthenticateMessageParams memory amp = params.deserializeAuthenticateMessageParams();
-        MarketTypes.DealProposal memory proposal = amp.message.deserializeDealProposal();
+        MarketTypes.DealProposal memory proposal = deserializeDealProposal(amp.message);
 
-        require(pieceToProposal[proposal.piece_cid].valid, "piece cid must be added before authorizing");
-        require(!pieceProviders[proposal.piece_cid].valid, "deal failed policy check: provider already claimed this cid");
+        require(pieceToProposal[proposal.piece_cid.data].valid, "piece cid must be added before authorizing");
+        require(!pieceProviders[proposal.piece_cid.data].valid, "deal failed policy check: provider already claimed this cid");
     }
 
     function dealNotify(bytes memory params) internal {
+        MarketDealNotifyParams memory mdnp = deserializeMarketDealNotifyParams(params);
+        MarketTypes.DealProposal memory proposal = deserializeDealProposal(mdnp.dealProposal);
 
-        MarketTypes.MarketDealNotifyParams memory mdnp = params.deserializeMarketDealNotifyParams();
-        MarketTypes.DealProposal memory proposal = mdnp.dealProposal.deserializeDealProposal();
-
-        require(pieceToProposal[proposal.piece_cid].valid, "piece cid must be added before authorizing");
-        require(!pieceProviders[proposal.piece_cid].valid, "deal failed policy check: provider already claimed this cid");
-
-        pieceProviders[proposal.piece_cid] = ProviderSet(proposal.provider, true);
-        pieceDeals[proposal.piece_cid] = mdnp.dealId;
+        require(pieceToProposal[proposal.piece_cid.data].valid, "piece cid must be added before authorizing");
+        require(!pieceProviders[proposal.piece_cid.data].valid, "deal failed policy check: provider already claimed this cid");
 
+        pieceProviders[proposal.piece_cid.data] = ProviderSet(proposal.provider.data, true);
+        pieceDeals[proposal.piece_cid.data] = mdnp.dealId;
     }
 
     // client - filecoin address byte format
-    function addBalance(bytes memory client, uint256 value) public {
-
+    function addBalance(CommonTypes.FilAddress memory client, uint256 value) public {
```

Comment:
Not critical for the demo or deal making flow but this API needs to remove the first arg as per the comments below. This was added for ease of testing before the market APIs were adapted